packages/xpath/src/evaluator/functions/FunctionImplementation.ts

```diff
@@ -127,6 +127,12 @@ export class FunctionImplementation<Length extends number> {
 							max: Infinity,
 						};
 
+					case 'variadic+':
```

Comment:
My naive outsider expectation would be that the parameter would still be `variadic` and that the implementation of the function would explicitly throw (or maybe eventually return an error result) for the 0 case. It feels to me not like the parameter type is not variadic but like the specific function has an additional requirement.

Something related that comes to mind is `indexed-repeat`. It needs to be given a path first and then pairs of paths and indexes. That means it is not allowed to have an even number of total parameters. I'm pretty sure an exception will be thrown if pairs aren't supplied https://github.com/getodk/web-forms/blob/9d83384cc10fee25ccfe2b5fd60cae62cde6481d/packages/xpath/src/functions/xforms/node-set.ts#L127 It feels similar to me as here because there's a function-specific constraint on "variadic".

JavaRosa makes these checks explicit: https://github.com/getodk/javarosa/blob/0d6c2eaa3704d28a5be4c3fc723baf56725f7bff/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java#L182 and https://github.com/getodk/javarosa/blob/0d6c2eaa3704d28a5be4c3fc723baf56725f7bff/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java#L498 That does mean the user can get a specific, helpful error message.